sdks/python/apache_beam/utils/subprocess_server.py

```diff
@@ -114,17 +114,27 @@ def start_process(self):
         cmd = [arg.replace('{{PORT}}', str(port)) for arg in self._cmd]
       endpoint = 'localhost:%s' % port
       _LOGGER.info("Starting service with %s", str(cmd).replace("',", "'"))
+
+      with tempfile.NamedTemporaryFile(delete=False) as stdout_file:
+        self._stdout_file_name = stdout_file.name
```

Comment:
From the [tempfile](https://docs.python.org/3/library/tempfile.html) docs:
> The resulting object can be used as a context manager (see [Examples](https://docs.python.org/3/library/tempfile.html#tempfile-examples)). On completion of the context or destruction of the file object the temporary file will be removed from the filesystem.

Doesn't this mean that the temp file will get deleted when this context completes? I think instead we can just not use it in a context, and the file will get deleted when stdout_file is destroyed:
```suggestion
      self._stdout_file = tempfile.NamedTemporaryFile(delete=False)
```

Reply:
I don't think the file  will be deleted since we set "delete=False" but probably safer to not use a context manager here.